lib/registration.pm

```diff
@@ -1109,4 +1111,34 @@ sub runtime_registration {
     zypper_call 'repos --details';
 }
 
+sub detect_suseconnect_path {
+    my $ret = '';
+
+    if (script_run("test -f /etc/SUSEConnect") == 0) {
+        $ret = '/etc/SUSEConnect';
+    } elsif (script_run("test -f /etc/SUSEConnect.example") == 0) {
+        $ret = '/etc/SUSEConnect.example';
+    } else {
+        die("Neither /etc/SUSEConnect nor /etc/SUSEConnect.example exist, please check!");
+    }
+
+    return $ret;
+}
+
+sub detect_scc_credentials_path {
+    my $ret = '';
+
+    if (script_run("test -f /etc/zypp/credentials.d/SCCcredentials") == 0) {
+        $ret = '/etc/zypp/credentials.d/SCCcredentials';
+    } elsif (script_run("test -f /run/secrets/SCCcredentials") == 0) {
+        $ret = '/run/secrets/SCCcredentials';
+    } elsif (script_run("test -f /run/secrets/credentials.d/SCCcredentials") == 0) {
+        $ret = '/run/secrets/credentials.d/SCCcredentials';
```

Comment:
Is it even possible that `/run/secrets` will be populated when the creds are missing in `/etc/zypp/` ?